samples/Mvc.LocalizationSample.Web/NotEqualToAttributeAdapter.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.DataAnnotations.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Mvc.LocalizationSample.Web
+{
+    internal class NotEqualToAttributeAdapter : AttributeAdapterBase<NotEqualToAttribute>
+    {
+        public NotEqualToAttributeAdapter(NotEqualToAttribute attribute, IStringLocalizer stringLocalizer)
+            : base(attribute, stringLocalizer)
+        {
+        }
+
+        public override void AddValidation(ClientModelValidationContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            MergeAttribute(context.Attributes, "data-val", "true");
```

Comment:
There is no JavaScript and therefore no client-side validation in the existing samples, this is just to make sure that anybody looking at this knows that that's the pattern. I've refactored it a bit to give a simpler idea of what you should do at the MergeAttribute part.